src/components/dashboard/containers/topNav/index.js

```diff
@@ -16,47 +17,84 @@ import {
     StyledTopNavLinkRight,
     StyledPrice,
     StyledAccount,
-    StyledSeparator
+    StyledSeparator,
+    StyledAccountInfo
 } from "./styles";
 
 class TopNav extends React.Component {
+    constructor(props) {
+        super(props);
+        this.toggleAccInfo = this.toggleAccInfo.bind(this);
+    }
+
     componentDidMount() {
         augmintTokenProvider();
         ratesProvider();
     }
+
+    toggleAccInfo(e) {
+        e.preventDefault();
+        e.stopPropagation();
+        this.props.toggleAccInfo();
+    }
+
     render() {
         const shortAddress = shortAccountAddresConverter(this.props.userAccount.address);
         const { ethBalance, tokenBalance } = this.props.userAccount;
+        const accountInfoData = {
+            account: this.props.userAccount,
+            rates: this.props.rates,
+            web3Connect: this.props.web3Connect
+        };
         return (
-            <StyledTopNav className={this.props.className}>
+            <StyledTopNav className={this.props.showAccInfo ? this.props.className + " hidden" : this.props.className}>
                 <TitleWrapper id="page-title" />
                 <StyledTopNavUl>
                     <StyledTopNavLi>
                         <StyledPrice>
                             <span className="price">€/ETH {this.props.rates.info.ethFiatRate}</span>
                         </StyledPrice>
                     </StyledTopNavLi>
-                    <StyledTopNavLi className="account">
-                        <StyledTopNavLinkRight title="Your account" to="/account" className="accountDetails">
-                            <Icon name="account" className="accountIcon" />
-                            <StyledPrice>
+                    <StyledTopNavLi className={this.props.showAccInfo ? "account" : "account hidden"}>
+                        <StyledTopNavLinkRight
+                            title="Your account"
+                            to={this.props.showAccInfo ? "" : "/account"}
+                            onTouchStart={this.toggleAccInfo}
```

Comment:
are you sure you need the `onTouchStart` event handler? Touch devices are usually understands the notion of `onClick`